tests/functional/snowflake_view_dependency/test_snowflake_view_dependency.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+
+_MODELS__DEPENDENT_MODEL_SQL = """
+
+{% if var('dependent_type', 'view') == 'view' %}
+    {{ config(materialized='view') }}
+{% else %}
+    {{ config(materialized='table') }}
+{% endif %}
+
+select * from {{ ref('base_table') }}
+
+"""
+
+
+
+_MODELS__BASE_TABLE_SQL = """
+
+{{ config(materialized='table') }}
+
+select *
+    {% if var('add_table_field', False) %}
+        , 1 as new_field
+    {% endif %}
+
+from {{ ref('people') }}
+
+"""
+
+_SEEDS__PEOPLE_CSV = """id,name
+1,Drew
+2,Jake
+3,Connor
+"""
+
+class TestSnowflakeLateBindingViewDependency:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dependent_model.sql":  _MODELS__DEPENDENT_MODEL_SQL,
```

Comment:
Q: is the _ prefix a thing we're doing on these variables or just something you're rolling for  this PR? No real problem with it -- just asking.

Reply:
Right on! I'll join in.